src/guide/advanced/queries.md

```diff
@@ -0,0 +1,385 @@
+# Queries
+
+Although much of the information about the state of the blockchain can be obtained, as we've shown before, using an event subscriber and using a filter to narrow the scope of the events to those of interest, sometimes, one needs to take a more direct approach. Enter queries. They are small instruction-like objects which, when sent to an Iroha peer, prompt a response with details from the current world state view.
+
+::: info
+
+As of writing, most queries can not be filtered, but can be paginated. As such, some queries, which we'll mark with a warning sign, should be used with care, and you should think carefully about the pagination scheme. 
+
+:::
+
+In the following section we shall try to mirror the module structure of the queries and present to you what they do.
+
+Before we proceed we should discuss a few conventions. We use the expressions _gets_, _returns_, _searches_ with a precise meaning in the following (somewhat encyclopædic) section.
+- _gets_, means that the query already has the data readily available and the data is trivial. Use these queries at will;
+- _returns_, means that the query has the data readily available, just as with _gets_, but by contrast the data is not trivial. You can still use these queries, but should be mindful not to ask for more information than you need;
+- _searches_, differs from the above two. Data must be actively collected and neither the return type, nor the collection process are cheap. These queries should be used **extremely sparingly** regardless of whether they have a corresponding warning about data volume or not;
+
+Another convention that we follow is that the queries are provided with just one data type as input, and parameterised by the type of the output. We should take some time to explain how to interpret the data.
+
+## How to read `FindZByXAndY` queries
+
+The queries will have a **Parameters** and a **Returns** section. The parameters can either be single or multiple types, while the output is almost always either one type, or a `Vec<Type>` kind of construction. 
+
+When we say "**Parameters**: (X, Y)", we mean that in Rust source code you need to construct the query as follows: 
+
+```rust
+let query = FindZByXAndY::new(x, y);
+```
+
+where `x` is a variable of type `X`, and `y` — `Y` respectively. Where necessary we'll provide you with information about each type, otherwise, refer to the guide for each programming language for more information. 
+
+When we say, "**Returns**: Vec<Z>", we mean that the return value is a collection of more than one element of that type. Depending on the SDK implementation this can be a type native to the language (e.g. JavaScript) or a thin wrappter around the Rust `Vec` structure. 
```

Comment:
When we say, "**Returns**: Vec<Z>", we mean that the return value is a collection of more than one element of that type. Depending on the SDK implementation this can be a type native to the language (e.g. JavaScript) or a thin wrapper around the Rust `Vec` structure.